### DISCUSSION TITLE
Data Provider and Virtual Pane

### DISCUSSION THREAD

**Discussion by 6pac** [6pac/SlickGrid#791] [Ideas]
@ghiscoding Just to let you know, I'm working on two decent sized PRs. 

The first is the Data Provider. Not so much that I desperately want in in the grid (I'm still using my own fork which got the DataProvider about 8 years ago), but I thought it would be good to get it out there so you can evaluate the approach and see if it suits your idea of what a Data Provider should be.

The second is a 'virtual pane' mod, which ideally should abstract away all the logic around the split/frozen panes and allow the grid code to be simplfied to more like it was before. As a side effect, the virtual pane will allow more complex dual H and V splits (up to 9 sub panes). Also, it will only create as many panes as are needed, so if no splits will ever be required for a grid, it will only create a single pane. This will reduce the number of HTML elements on the page, with speed and clutter reduction benefits (since I've never wanted to split a grid, or used the grid split feature, this makes sense to me).
It's early days - I've only put together the basic structure and it's yet to be seen how messy this will get, but I'm giving it a decent go.

These are all 'backburner' projects so they may take many months to come together. But just thought it was a good idea to mention this stuff.

**Comment by ghiscoding** [comment]
Oh good, I'm sure there better ways than current implementation of frozen grid, it would be good to see and test your approach. I'm not sure if you were expecting to ship this under the current version or under the next major though? Are there any breaking changes to your new code? If there are then maybe wait for the code restructuring that I'm doing for the next major, I'll be moving all files into a `src/` folder to support TypeScript and of course the build will go under `dist/`, technically speaking the users will no longer be able to use the source file directly, they'll always have to use the built files from `dist/` in the future.

**Reply by 6pac** [reply]
I'm not putting any time constraints on this - it's ready when it's ready. Releases should proceed independently.

The Data Provider may be breaking, I'm not sure yet, haven't started (but the code already exists, I'm just porting it between forks).
The Virtual Pane feature will be specifically 100% non breaking for our code and examples (that's kind of the point), but since the underlying grid HTML structure could potentially change (with panes no longer generated if not needed), it should be classified as a breaking change. There may also be minor breaking changes to the frozen pane configuration options to accommodate the larger feature set. Again, we'll see and this is open to negotiation.
